src/cgame/rocket/rocketDataSelect.h

```diff
@@ -85,35 +86,36 @@ class RocketDataSelect : public Rml::ElementFormControlDataSelect, public Rml::E
 	{
 		extern std::queue< RocketEvent_t * > eventQueue;
 
-		if ( event.GetTargetElement() == owner && event == "show" )
+		if ( event == Rml::EventId::Show )
 		{
-			eventQueue.push( new RocketEvent_t( this, va( "setDataSelectValue %s %s", dsName.c_str(), tableName.c_str() ) ) );
-		}
-	}
+			ASSERT_EQ( event.GetTargetElement(), owner );
 
-	void OnUpdate()
-	{
-		extern std::queue< RocketEvent_t * > eventQueue;
+			if ( !selectionInit )
+			{
+				selectionInit = true;
+				int index = CG_Rocket_GetDataSourceIndex( dsName.c_str(), tableName.c_str() );
 
-		ElementFormControlDataSelect::OnUpdate();
+				if ( index >= 0 )
+				{
+					SetSelection( index );
+				}
 
-		if ( GetSelection() != selection )
+				// only add this after initializing the selection so events won't pingpong
+				this->AddEventListener( Rml::EventId::Change, this );
+			}
+		}
+		else if ( event == Rml::EventId::Change )
 		{
-			selection = GetSelection();
+			ASSERT_EQ( event.GetTargetElement(), this );
 
 			// dispatch event so cgame knows about it
-			eventQueue.push( new RocketEvent_t( Rml::String( va( "setDS %s %s %d", dsName.c_str(), tableName.c_str(), selection ) ) ) );
-
-			// dispatch event so rocket knows about it
-			Rml::Dictionary parameters;
-			parameters[ "index" ] = va( "%d", selection );
-			parameters[ "datasource" ] = dsName;
-			parameters[ "table" ] = tableName;
-			DispatchEvent( "rowselect", parameters );
+			eventQueue.push( new RocketEvent_t( Rml::String(
+				va( "setDS %s %s %d", dsName.c_str(), tableName.c_str(), GetSelection() ) ) ) );
```

Comment:
Rml::String is now stdstring right? So you might as well use Str::Format here?

Reply:
Done.